### PR TITLE
To add feature 2

### PR DESCRIPTION
**Pull request by Araujo-Raiara**

### CODE REVIEW

File: app/src/main/java/com/victor/adasfeed/FeedActivity.kt [hunk 1]
```diff
@@ -92,7 +92,7 @@ class FeedActivity : AppCompatActivity() {
         storiesLayoutManager.orientation = LinearLayoutManager.HORIZONTAL
 
         storiesAdapter = StoriesAdapter()
-        postAdapter = PostAdapter()
+        postAdapter = PostAdapter(clickUndoDelete = ::onUndoDelete)
```
Comment (VictorMagosso) [inline app/src/main/java/com/victor/adasfeed/FeedActivity.kt:95]: 🚀

File: app/src/main/res/layout/post_item.xml [hunk 2]
```diff
@@ -37,36 +37,37 @@
...
-        android:layout_width="16dp"
-        android:layout_height="16dp"
-        android:layout_marginLeft="14dp"
+    <com.google.android.material.button.MaterialButton
+        android:id="@+id/liked_button"
+        style="@style/ButtonWithIconOnly"
+        app:icon="@drawable/ic_empty_liked"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="14dp"
```
Comment (joaovq) [inline app/src/main/res/layout/post_item.xml:46]: This could'nt be in dimen.xml?

Comment (Araujo-Raiara) [reply app/src/main/res/layout/post_item.xml:46]: done

Comment (joaovq) [reply app/src/main/res/layout/post_item.xml:46]: 🚀

File: app/src/main/java/com/victor/adasfeed/PostAdapter.kt [hunk 3]
```diff
@@ -86,70 +86,88 @@ class PostAdapter(private val postList: MutableList<Post> = mockedPostList()) :
...
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
 
-        holder.imageLiked.setOnClickListener {
-            holder.imageLiked.setImageResource(R.drawable.liked_fill)
+        }
+
+
+        holder.imagePost.setOnLongClickListener { it ->
```
Comment (joaovq) [inline app/src/main/java/com/victor/adasfeed/PostAdapter.kt:117]: 👏

File: app/src/main/java/com/victor/adasfeed/PostAdapter.kt [hunk 4]
```diff
@@ -86,70 +86,88 @@ class PostAdapter(private val postList: MutableList<Post> = mockedPostList()) :
     override fun getItemCount() = postList.size
 
     override fun onBindViewHolder(holder: PostViewHolder, position: Int) {
-        holder.textUserName.text = postList[position].userName
-        holder.textPostDescription.text = postList[position].description
-        holder.imagePost.setImageResource(postList[position].imagePost)
-        holder.imageUser.setImageResource(postList[position].imageUser)
+        val currentPost = postList[position]
+        holder.textUserName.text = currentPost.userName
```
Comment (VictorMagosso) [inline app/src/main/java/com/victor/adasfeed/PostAdapter.kt:90]: We could use `apply {}` or `with(postList[position]) {}`

File: app/src/main/java/com/victor/adasfeed/PostAdapter.kt [hunk 5]
```diff
@@ -86,70 +86,88 @@ class PostAdapter(private val postList: MutableList<Post> = mockedPostList()) :
...
+        if (currentPost.isLiked) {
+            holder.likedButton.setIconResource(R.drawable.ic_liked)
+        } else {
+            holder.likedButton.setIconResource(R.drawable.ic_empty_liked)
+        }
+        holder.likedButton.setOnClickListener {
+            val newObject = currentPost.copy(isLiked = !currentPost.isLiked)
+            val newList = ArrayList(postList)
+            newList[position] = newObject
+            setNewList(newList)
```
Comment (VictorMagosso) [inline app/src/main/java/com/victor/adasfeed/PostAdapter.kt:104]:
```kotlin
if (currentPost.isLiked) {
            holder.likedButton.setIconResource(R.drawable.ic_liked)
        } else {
            holder.likedButton.setIconResource(R.drawable.ic_empty_liked)
        }
```

File: app/src/main/java/com/victor/adasfeed/PostAdapter.kt [hunk 6]
```diff
@@ -86,70 +86,88 @@ class PostAdapter(private val postList: MutableList<Post> = mockedPostList()) :
...
+                }.setDuration(5_000).show()
+            true
         }
     }
 
-    fun addNewPost(newPost: Post) {
-        postList.add(newPost)
+    fun addNewPost(newPost: Post, position: Int? = null) {
+        if (position != null) {
+            postList.add(position, newPost)
```
Comment (VictorMagosso) [inline app/src/main/java/com/victor/adasfeed/PostAdapter.kt:129]:
```kotlin
position?.let { safePosition ->
postList.add(safePosition , newPost) 
} ?: postList.add(newPost)
```

File: app/src/main/java/com/victor/adasfeed/PostAdapter.kt [hunk 7]
```diff
@@ -86,70 +86,88 @@ class PostAdapter(private val postList: MutableList<Post> = mockedPostList()) :
...
         notifyItemInserted(postList.indexOf(newPost))
         // perde performance pois recria a lista inteira
         // o DiffUtilCallback é melhor
     }
 
-    fun setNewList(newList: List<Post>) {
+    private fun removePost(position: Int) {
+        val newPostList = ArrayList(postList)
+        newPostList.removeAt(position)
+        setNewList(newPostList)
```
Comment (VictorMagosso) [inline app/src/main/java/com/victor/adasfeed/PostAdapter.kt:141]:
```kotlin
notityItemRemoved(position)
notifyItemRangeChanged(position, newList.size)
```

File: app/src/main/java/com/victor/adasfeed/PostAdapter.kt [hunk 8]
```diff
@@ -86,70 +86,88 @@ class PostAdapter(private val postList: MutableList<Post> = mockedPostList()) :
...
     val imageFavorite: ImageView
+    val likedButton: MaterialButton
 
     init {
         textUserName = view.findViewById(R.id.textUserName)
         textPostDescription = view.findViewById(R.id.textPostDescription)
         imageUser = view.findViewById(R.id.imageUser)
         imagePost = view.findViewById(R.id.imagePost)
-        imageLiked = view.findViewById(R.id.imageLiked)
+        likedButton = view.findViewById(R.id.liked_button)
```
Comment (VictorMagosso) [inline app/src/main/java/com/victor/adasfeed/PostAdapter.kt:170]: 🚀 Nice pattern!

**Pull request closed without merging**